stdlib/runtime/numbers.gr

```diff
@@ -1156,6 +1156,74 @@ let numberAddSubBigIntHelp = (x, y, isSub) => {
   }
 }
 
+@unsafe
+export let addSubRational = (x, y, isSub, keepRational) => {
+  let xNumerator = boxedRationalNumerator(x)
+  let xDenominator = boxedRationalDenominator(x)
+  let yNumerator = boxedRationalNumerator(y)
+  let yDenominator = boxedRationalDenominator(y)
+  if (BI.eq(xDenominator, yDenominator)) {
+    let newNumerator =
+      if (isSub) BI.sub(xNumerator, yNumerator)
+      else BI.add(xNumerator, yNumerator)
+    let ret = reducedFractionBigInt(newNumerator, xDenominator, keepRational)
+    Memory.decRef(newNumerator)
+    ret
+  } else {
+    let numerator1 = BI.mul(xNumerator, yDenominator)
+    let numerator2 = BI.mul(yNumerator, xDenominator)
+    let numerator =
+      if (isSub) BI.sub(numerator1, numerator2)
+      else BI.add(numerator1, numerator2)
+    let denominator = BI.mul(xDenominator, yDenominator)
+    let ret = reducedFractionBigInt(numerator, denominator, keepRational)
+    Memory.decRef(numerator1)
+    Memory.decRef(numerator2)
+    Memory.decRef(numerator)
+    ret
+  }
+}
+
+@unsafe
+export let timesDivideRational = (x, y, isDivide, keepRational) => {
+  let xNumerator = boxedRationalNumerator(x)
+  let xDenominator = boxedRationalDenominator(x)
+  let yNumerator = boxedRationalNumerator(y)
+  let yDenominator = boxedRationalDenominator(y)
+  // (a / b) * (c / d) == (a * c) / (b * d)
+  // (a / b) / (c / d) == (a * d) / (b * c)
+  let numerator =
+    if (isDivide) BI.mul(xNumerator, yDenominator)
+    else BI.mul(xNumerator, yNumerator)
+  let denominator =
+    if (isDivide) BI.mul(xDenominator, yNumerator)
+    else BI.mul(xDenominator, yDenominator)
+  reducedFractionBigInt(numerator, denominator, keepRational)
+}
+
+@unsafe
+export let cmpRationals = (x, y) => {
```

Comment:
Would call this `cmpRational` to match the style of the others.

Reply:
I named in `cmpRationals` because `cmpRational` already exists (it compares a rational number to an unknown number type). I thought adding the `s` would make it clearer that it is known that both are rational